### ISSUE TITLE
max_value() vs MAX

### ISSUE THREAD

**Issue by snowe2010** [rust-lang/rust#29841]
I'm not sure if I'm calling this wrong, but when I try to use 

`let inf = std::u32::max_value();`

I get the error: 

```
src\main.rs:3:13: 3:32 error: unresolved name `std::u32::max_value` [E0425]
src\main.rs:3   let inf = std::u32::max_value();
```
, but following PR #11790 it looks like it should be 

`let inf = std::u32::MAX;`

which of course works. None of  this is a problem, except the documentation says to use max_value()

https://doc.rust-lang.org/std/primitive.u32.html

Please let me know if I'm doing something wrong though.

**Comment by Gankro** [comment]
The problem is that `u32::max_value` is on the u32 *type*, but `std::u32` is the u32 *module*.

```rust
let inf = u32::max_value();
```

works.

**Comment by snowe2010** [comment]
oh great! Thank you. I guess I'll close this then.

**Comment by Gankro** [comment]
This is definitely a problematic thing, though. @steveklabnik do you think there's anything that can be done here?

**Comment by bluss** [comment]
Can we remove the std::u32 module and its relatives? It's so useless.. http://doc.rust-lang.org/nightly/std/u32/index.html

Maybe just hide it?

**Comment by Gankro** [comment]
It's waiting on associated consts, iirc.

**Comment by steveklabnik** [comment]
We link to the module from primitive pages and vice versa, so I'm not really sure there's much to do. I mean, other than add examples, which will happen as part of https://github.com/rust-lang/rust/issues/29358

**Comment by Gankro** [comment]
SGTM, closing in favour of that.
